src/toctype.c

```diff
@@ -182,15 +182,21 @@ class ToCtypeVisitor : public Visitor
     void visit(TypeEnum *t)
     {
         //printf("TypeEnum::toCtype() '%s'\n", t->sym->toChars());
-        Type *tm = t->mutableOf();
-        if (tm->ctype && tybasic(tm->ctype->Tty) == TYenum)
+        if (t->sym->memtype->toBasetype()->ty == Tint32)
```

Comment:
Out of curiosity, is this a dmd limitation?  You should be able to make all integral types (but not bool) as enums.